src/cli/cli.cpp

```diff
@@ -3758,9 +3762,240 @@ void Interpreter::HandleDiagnosticGetResponse(Message &aMessage, const Ip6::Mess
+            mServer->OutputFormat("\r\n");
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_SHORT_ADDRESS:
+            mServer->OutputFormat("Rloc16: 0x%04x\r\n", diagTlv.mAddr16);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_MODE:
+            mServer->OutputFormat("Mode:\r\n");
+            OutputMode(diagTlv.mMode, column + INDENT_SIZE);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_POLLING_PERIOD:
+            mServer->OutputFormat("Timeout: %u\r\n", diagTlv.mTimeout);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_CONNECTIVITY:
+            mServer->OutputFormat("Connectivity:\r\n");
+            OutputConnectivity(diagTlv.mConnectivity, column + INDENT_SIZE);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_ROUTE:
+            mServer->OutputFormat("Route:\r\n");
+            OutputRoute(diagTlv.mRoute, column + INDENT_SIZE);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_LEADER_DATA:
+            mServer->OutputFormat("Leader Data:\r\n");
+            OutputLeaderData(diagTlv.mLeaderData, column + INDENT_SIZE);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_NETWORK_DATA:
+            mServer->OutputFormat("Network Data: ");
+            OutputBytes(diagTlv.mNetworkData, diagTlv.mNetworkDataCount);
+            mServer->OutputFormat("\r\n");
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_IP6_ADDR_LIST:
+            mServer->OutputFormat("IP6 Address List:\r\n");
+            for (uint16_t i = 0; i < diagTlv.mIp6AddrCount; ++i)
+            {
+                OutputSpaces(column + INDENT_SIZE);
+                mServer->OutputFormat("- ");
+                OutputIp6Address(diagTlv.mIp6AddrList[i]);
+                mServer->OutputFormat("\r\n");
+            }
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_MAC_COUNTERS:
+            mServer->OutputFormat("MAC Counters:\r\n");
+            OutputNetworkDiagMacCounters(diagTlv.mMacCounters, column + INDENT_SIZE);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_BATTERY_LEVEL:
+            mServer->OutputFormat("Battery Level: %u%%\r\n", diagTlv.mBatteryLevel);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_SUPPLY_VOLTAGE:
+            mServer->OutputFormat("Supply Voltage: %umV\r\n", diagTlv.mSupplyVoltage);
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_CHILD_TABLE:
+            mServer->OutputFormat("Child Table:\r\n");
+            for (uint16_t i = 0; i < diagTlv.mChildCount; ++i)
+            {
+                OutputSpaces(column + INDENT_SIZE);
+                mServer->OutputFormat("- ");
+                OutputChildTableEntry(diagTlv.mChildTable[i], column + INDENT_SIZE + 2);
+            }
+            break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_CHANNEL_PAGES:
+            mServer->OutputFormat("Channel Pages: ");
+            OutputBytes(diagTlv.mChannelPages, diagTlv.mChannelPageCount);
```

Comment:
OutputBytes prints unquoted hex bytes. If loaded as yaml, the content could be parsed either string or int, depending on whether the value is an 10 based integer.  I think we should always output quoted hex bytes so that the yaml can always decode the value as string.